tests/fetchODCDataFilesUtilsTests.js

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright 2020 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in compliance with this licence.
+You may obtain a copy of the BSD License at
+https://raw.githubusercontent.com/inclusive-design/data-update-github/main/LICENSE
+*/
+
+"use strict";
+
+const fluid = require("infusion");
+const jqUnit = fluid.require("node-jqunit", require, "jqUnit");
+const fs = require("fs");
+const nock = require("nock");
+const rimraf = require("rimraf");
+
+const utils = require("../scripts/fetchODCDataFilesUtils.js");
+
+jqUnit.module("Fetch ODC data files tests");
+
+const odsHostname = "https://data.ontario.ca";
+const odsPath = "/dataset/covid-19-assessment-centre-locations";
+const fullSitePath = odsHostname + odsPath;
+
+//****************** Test getDataSource() ******************
+
+const getDataSourceTestCases = {
+	hasCsv: {
+		message: "The response contains a download link pointing to a csv format: download link and the published date is returned",
+		nockConfig: {
+			status: 200,
+			response: fs.readFileSync(__dirname + "/data/responseWithCsv.html")
+		},
+		expected: {
+			downloadUrl: "https://data.ontario.ca/dataset/8ba078b2-ca9b-44c1-b5db-9674d85421f9/resource/c60993bb-3988-4648-9be9-398dee480514/download/assessment_centre_locations.csv",
+			publishedDate: "2020-11-17"
+		}
+	},
+	noCsv: {
+		message: "The response doesn't contain a download link pointing to a csv format: download link and the published date returns undefined",
+		nockConfig: {
+			status: 200,
+			response: fs.readFileSync(__dirname + "/data/responseWithoutCsv.html")
+		},
+		expected: {
+			downloadUrl: undefined,
+			publishedDate: undefined
+		}
+	},
+	serverDown: {
+		message: "No respose from the server",
+		nockConfig: {
+			status: 400,
+			response: ""
+		},
+		expected: {
+			downloadUrl: undefined,
+			publishedDate: undefined
+		}
+	}
+};
+
+fluid.each(getDataSourceTestCases, function (oneCase, key) {
+	jqUnit.asyncTest("Test getDataSource() - " + key, function () {
```

Comment:
If you update to 3.0.0-dev.20201113T133151Z.fdbc62672.FLUID-6145 you should be able to use the new idiomatic support for promise-based tests - e.g.

    jqUnit.test("Test getDataSource() - " + key, function () {
        return utils.getDataSource(fullSitePath).then(function (result) {
			jqUnit.assertDeepEq(oneCase.message, oneCase.expected, result);
    };
)

Reply:
@amb26, the suggested change doesn't work.

After some digging, it seems 3.0.0-dev.20201113T133151Z.fdbc62672.FLUID-6145 release doesn't include your latest jqUnit support for promise based tests.

If you search "FLUID-6577" in [the jqUnit.js at the commit hash fdbc62672](https://github.com/amb26/infusion/blob/fdbc62672406586c2399d213a2e4e1909334d7bc/tests/test-core/jqUnit/js/jqUnit.js), nothing will be found.

I was expecting jqUnit.js changes in this merge: https://github.com/fluid-project/infusion/commit/1619f14b960ccf2727db0db45570811186a26aeb

Let me know if I missed anything. Thanks.

Reply:
Very sorry, the release seemed to be from the right date since I couldn't think of another reason for a release around that time. I've republished 3.0.0-dev.20201113T153152Z.32176dcbe.FLUID-6145 from a commit 2 hours later which does include the fix!

Reply:
Thanks for pointing out, @amb26. This release works perfectly. I should have looked more carefully thru the infusion release list. ;)

Reply:
No, it was entirely my fault - fluid-publish dates the release from the commit rather from the current time and the release didn't exist at the time you looked for it. Somehow at the time I had managed to make the release from the previous commit.